assets/js/data.coffee

```diff
@@ -249,11 +263,65 @@ class Data
   toggleCollapsed: (row) ->
     @store.setCollapsed row.id, (not @collapsed row)
 
+  countInstances: (id) ->
+    # Precondition: No circular references in ancestry
+    errors.assert id?, "Empty id passed to countInstances"
+    if id == constants.root_id
+      return 1
+    parentCount = 0
+    for parent_id in (@store.getParents id)
+      parentCount += @countInstances parent_id # Always exactly once under every parent
+    return parentCount
```

Comment:
Fined-tuned countInstances. exactlyOneInstance now has optimal complexity without maintaining extra data. (you could argue this is a little more complicated to read, but I think